src/pages/[...slug].tsx

```diff
@@ -433,6 +437,9 @@ const ContentPage: NextPage<ContentPageProps> = ({
         handle: seo?.twitter_creator,
         site: seo?.twitter_site,
     };
+    let canonicalUrl = seo?.canonical_url
+        ? seo?.canonical_url
+        : publicRuntimeConfig.absoluteBasePath + router.asPath;
```

Comment:
this is believe is self referencing if they don't explicitly add any SEO in strapi.

Reply:
Correct, `publicRuntimeConfig.absoluteBasePath + router.asPath` will be the existing page (self referencing value).